mllib/src/main/scala/org/apache/spark/mllib/linalg/distributed/RowMatrix.scala

```diff
@@ -775,6 +777,34 @@ class RowMatrix @Since("1.0.0") (
         s"The number of rows $m is different from what specified or previously computed: ${nRows}.")
     }
   }
+
+  /**
+    * Computing desired tree aggregate depth necessary to avoid exceeding
+    * driver.MaxResultSize during aggregation.
+    * Based on the formulae: (numPartitions)^(1/depth) * objectSize <= DriverMaxResultSize
+    * @param aggregatedObjectSizeInMb the size, in megabytes, of the object being tree aggregated
+    * @param numPartitions the number of partitions for which to aggregate partial results
+    * @param maxDriverResultSizeInMb the value of the parameter spark.driver.maxResultSize in MB
+    */
+  private[spark] def getTreeAggregateIdealDepth(
+      aggregatedObjectSizeInMb: Int,
+      numPartitions: Int = rows.getNumPartitions,
+      maxDriverResultSizeInMb: Long = rows.conf.get[Long](MAX_RESULT_SIZE)) = {
+    logWarning(
```

Comment:
Yeah, it's a fair point, failing earlier. I think it is OK and simpler to not plumb through an extra arg here to check it here too. It will fail fairly quickly. I don't feel strongly about it but don't know if we need to bother checking it here at all.

Also can you get rid of numPartitions? you can create an empty or dummy RDD in the test with as many partitions as you like. It should always be using rows.getNumPartitions anyway I think.

Reply:
Okay to get rid of numPartitions argument.
But with the same logic in mind, I guess I should I get rid of the maxDriverResultSizeInMb argument aswell, since we can access it from sparkConf.
Will do that in next update, to come soon.

Reply:
Yeah, that's why I was saying just don't bother with the check at all.